base/test/filewritermodule_tests.cpp

```diff
@@ -52,4 +52,62 @@ BOOST_AUTO_TEST_CASE(basic, * boost::unit_test::disabled())
 	delete[] pReadData;
 }
 
+BOOST_AUTO_TEST_CASE(append)
+{
+	const uint8_t* pReadData = nullptr;
+	unsigned int readDataSize = 0U;
+
+	auto m1 = boost::shared_ptr<ExternalSourceModule>(new ExternalSourceModule());
+	auto metadata = framemetadata_sp(new FrameMetadata(FrameMetadata::ENCODED_IMAGE));
+	auto pinId = m1->addOutputPin(metadata);
+	auto m2 = boost::shared_ptr<FileWriterModule>(new FileWriterModule(FileWriterModuleProps("./data/testOutput/fileWriterModuleSample.txt", true)));
+	m1->setNext(m2);
+
+	BOOST_TEST(m1->init());
+	BOOST_TEST(m2->init());
+	const char *stringToAppend = "Apra";
+	readDataSize = strlen(stringToAppend);
+	auto frame = m1->makeFrame(readDataSize, pinId);
+	memcpy(frame->data(), stringToAppend, readDataSize);
+
+
+	frame_container frames;
+	frames.insert(make_pair(pinId, frame));
+	for (auto i = 0; i < 4; i++)
+	{
+		m1->send(frames);
+		m2->step();
+	}
```

Comment:
1. There should be some assertion (BOOST_TEST) to check which file got created/appended, this test does not prove that the feature works.
2. The file(s) created during the test should be deleted in a teardown function 
For an example of how to use test fixtures with setup and teardown see here:
https://www.boost.org/doc/libs/1_79_0/libs/test/doc/html/boost_test/tests_organization/fixtures/case.html